File: benchmarks/fixtures/webapp_rb/models/session.rb
# Session model.

class Session
  attr_reader :token, :user, :user_id, :expires_at

  def initialize(user:, token:, expires_at:)
    @user = user
    @user_id = user.id
    @token = token
    @expires_at = expires_at
  end

  def expired?
    Time.now.utc > @expires_at
  end

  def delete
    # Remove session from storage
  end

  def self.create(user:, token:, expires_in:)
    expires_at = Time.now.utc + expires_in
    new(user: user, token: token, expires_at: expires_at)
  end

  def self.find_by_token(token)
    # Look up session by token
  end

  def self.find_all_by_user(_user)
    # Find all sessions for a user
    []
  end
end

File: benchmarks/fixtures/webapp_php/models/session.php
<?php

namespace App\Models;

/**
 * Session model.
 */
class Session
{
    public function __construct(
        public readonly User $user,
        public readonly string $token,
        public readonly int $expiresAt,
    ) {
    }

    public function expired(): bool
    {
        return time() > $this->expiresAt;
    }

    public function delete(): void
    {
        // Remove session from storage.
    }

    public static function create(User $user, string $token, int $expiresIn): self
    {
        return new self($user, $token, time() + $expiresIn);
    }

    public static function findByToken(string $token): ?self
    {
        return null;
    }
}
